src/odinapi/utils/smrl2filewriter.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+import os
+import datetime as dt
+import argparse
+import attr
+from typing import List, Dict, Any
+from dateutil.relativedelta import relativedelta
+from netCDF4 import Dataset, date2num
+
+from odinapi.database import level2db
+from odinapi.views.get_ancillary_data import get_ancillary_data
+from odinapi.views.database import DatabaseConnector
+from odinapi.utils import datamodel
+
+
+@attr.s
+class L2Getter:
+    freqmode = attr.ib(type=int)
+    product = attr.ib(type=str)
+    db1 = attr.ib(type=DatabaseConnector)
+    db2 = attr.ib(type=level2db.Level2DB)
+
+    def get_l2i(self, scanid: int) -> datamodel.L2i:
+        return datamodel.to_l2i(self.db2.get_L2i(self.freqmode, scanid))
+
+    def get_l2anc(self, l2: Dict[str, Any]) -> datamodel.L2anc:
+        return datamodel.to_l2anc(get_ancillary_data(self.db1, [l2])[0])
+
+    def get_l2full(self, scanid: int) -> datamodel.L2Full:
+        l2dict = self.db2.get_L2(self.freqmode, scanid, self.product)[0]
+        return datamodel.L2Full(
+            l2i=self.get_l2i(scanid),
+            l2anc=self.get_l2anc(l2dict),
+            l2=datamodel.to_l2(l2dict, self.product)
+        )
+
+    def get_data(self, scanids: List[int]) -> List[datamodel.L2Full]:
+        l2fulls = [self.get_l2full(scanid) for scanid in scanids]
+        return [l2full for l2full in l2fulls if l2full.l2i.isvalid()]
+
+    def get_scanids(self, start: dt.datetime, end: dt.datetime) -> List[int]:
+        scans = []
+        while start < end:
+            currentscans = self.db2.get_scans(
+                self.freqmode,
+                start_time=start,
+                end_time=start + relativedelta(days=1)
+            )
+            for scan in currentscans:
+                scans.append(scan["ScanID"])
+            start += relativedelta(days=1)
+        return scans
+
+
+@attr.s
+class L2FileCreater:
+    project = attr.ib(type=str)
+    freqmode = attr.ib(type=int)
+    product = attr.ib(type=str)
+    parameters = attr.ib(type=datamodel.L2File)
+    data = attr.ib(type=List[datamodel.L2Full])
+    outdir = attr.ib(type=str)
+
+    @property
+    def start(self) -> dt.datetime:
+        return self.data[0].l2.Time
+
+    @property
+    def end(self) -> dt.datetime:
+        return self.data[-1].l2.Time
+
+    @property
+    def ntimes(self) -> int:
+        return len(self.data)
+
+    @property
+    def nlevels(self) -> int:
+        return len(self.data[0].l2.Profile)
+
+    @property
+    def invmode(self) -> str:
+        return self.data[0].l2.InvMode
+
+    def filename(self) -> str:
+        if not os.path.isdir(self.outdir):
+            os.makedirs(self.outdir)
```

Comment:
```suggestion
        os.makedirs(self.outdir, exist_ok=True)
```
(Mostly fyi, but also a little safer if two bulk extractions run i parallel)